src/Style.hs

```diff
@@ -16,23 +21,64 @@ data PropKey
     | BackgroundColor
   deriving (Eq, Ord, Show)
 
+data Units
+    = Px
+    | Em
+    | Percent
+    | Pt
+    deriving (Eq, Show)
+
 data PropVal
-    = Px Int
+    = NumUnit Int Units
     | Color SDL.Color
     | Auto
+    | None 
   deriving (Eq, Show)
 
--- TODO: Replace with real parser
-keyStr :: String -> PropKey
-keyStr "background-color" = BackgroundColor
-keyStr "width" = Width
+-- Parse the keys given in through a String
+parseKey :: String -> Maybe PropKey
+parseKey str = maybeResult (parse keyParser (pack str))
+
+-- Parse the values given in through a String
+parseVal :: String -> Maybe PropVal
+parseVal str = maybeResult (parse valParser (pack str))
```

Comment:
these two functions have the same shape. we should create a `parseFromText :: Parser a -> String -> Maybe a` that abstracts this out